java-manta-client-unshaded/src/main/java/com/joyent/manta/client/MantaClient.java

```diff
@@ -411,6 +411,341 @@ public boolean isClosed() {
+     * @return true when bucket was created
+     * @throws IOException                                     If an IO exception has occurred.
+     * @throws MantaClientHttpResponseException                If a http status code {@literal > 300} is returned.
+     */
+    public boolean createBucket(final String rawPath, final MantaHttpHeaders rawHeaders)
+            throws IOException {
+        Validate.notBlank(rawPath, "CREATE bucket path must not be empty nor null");
+
+        String path = formatPath(rawPath);
+
+        LOG.debug("CREATE    {} [bucket]", path);
+
+        final HttpPut put = httpHelper.getRequestFactory().put(path);
+
+        final MantaHttpHeaders headers;
+
+        if (rawHeaders == null) {
+            headers = new MantaHttpHeaders();
+        } else {
+            headers = rawHeaders;
+        }
+
+        MantaHttpRequestFactory.addHeaders(put, headers.asApacheHttpHeaders());
+
+        put.setHeader(HttpHeaders.CONTENT_TYPE, MantaContentTypes.BUCKET.getContentType());
+
+        HttpResponse response = httpHelper.executeAndCloseRequest(put,
+                HttpStatus.SC_NO_CONTENT,
+                "PUT    {} response [{}] {} ");
+
+        // When LastModified is set, the bucket already exists
+        return response.getFirstHeader(HttpHeaders.LAST_MODIFIED) == null;
+    }
+
+    /**
+     * Deletes a bucket in Manta with the given path.
+     * @param rawPath Path of the bucket you want to delete.
+     * @param requestHeaders requestHeaders HTTP headers to attach to request (may be null)
+     * @throws IOException Problem getting object over the network
+     */
+    void deleteBucket(final String rawPath, final MantaHttpHeaders requestHeaders)
+            throws IOException {
+        Validate.notBlank(rawPath, "rawPath must not be blank");
+        final String path = formatPath(rawPath);
+        LOG.debug("DELETE {} bucket", path);
+
+        try {
+            httpHelper.httpDelete(path, requestHeaders);
+        } catch (MantaClientHttpResponseException e) {
+
+            // Attempting to delete a non-empty bucket will result in an error
+            if (e.getServerCode().equals(MantaErrorCode.BUCKET_NOT_FOUND_ERROR)
+                    || e.getServerCode().equals(MantaErrorCode.BUCKET_NOT_EMPTY_ERROR)) {
+                final MantaIOException mioe = new MantaIOException("Unable to delete bucket", e);
+                mioe.setContextValue("bucket", path);
+                throw mioe;
+            }
+        } catch (ConnectionPoolTimeoutException e) {
+            final MantaClientException clientException = new MantaClientException(
+                    "Connection pool timeout while deleting bucket" + e.getMessage());
+            clientException.setContextValue("path", path);
+            throw clientException;
+
+        } catch (IOException e) {
+            throw new IOException(e);
+        }
+        LOG.debug("Finished deleting bucket {}.", path);
```

Comment:
Please format log message like all of the other log messages.

Reply:
Refactored.